Kernel/Memory/paging.cpp

```diff
@@ -226,21 +226,30 @@ static uintptr_t findNextFreeVirtualAddress(size_t seq)
 
 void* newPage(size_t size)
 {
+    void* page = nullptr;
     pagingLock.lock();
+
     size_t page_count = PAGE_COUNT(size);
     size_t free_idx = findNextFreeVirtualAddress(page_count);
-    if (free_idx == SIZE_MAX)
-        return NULL;
+    if (free_idx == SIZE_MAX) {
+        goto fail;
+    }
+
     for (size_t i = free_idx; i < free_idx + page_count; i++) {
         size_t phys_page_idx = physical.findNextFreePhysicalAddress();
-        if (phys_page_idx == SIZE_MAX)
-            return NULL;
+        if (phys_page_idx == SIZE_MAX) {
+            goto fail;
+        }
         Arch::Memory::Address phys(phys_page_idx * ARCH_PAGE_SIZE);
         Arch::Memory::Address vaddr(i * ARCH_PAGE_SIZE);
         mapKernelPage(vaddr, phys);
     }
+
+    page = (void*)(free_idx * ARCH_PAGE_SIZE);
+
+fail:
```

Comment:
This would be a good place to use RAII semantics to release the lock on return.

Reply:
Yeah, I was thinking that too. I was probably going to make a lambda function for mutexes and semaphores like we do for interrupt critical regions.

Reply:
Actually, what I'll probably do is write a wrapper class around Mutex that uses RAII. I could then write another wrapper class for lambda functionality, that way we have either one we want.

Reply:
Resolved by #370